repo-scripts/changelog-generator/index.ts

```diff
@@ -0,0 +1,110 @@
+import fetch from 'node-fetch';
+
+const changelogFunctions: ChangelogFunctions = {
+  getDependencyReleaseLine: async (
+    changesets,
+    dependenciesUpdated,
+    options
+  ) => {
+    if (!options.repo) {
+      throw new Error(
+        'Please provide a repo to this changelog generator like this:\n"changelog": ["@changesets/changelog-github", { "repo": "org/repo" }]'
+      );
+    }
+    if (dependenciesUpdated.length === 0) return '';
+
+    const changesetLink = `- Updated dependencies [${(
+      await Promise.all(
+        changesets.map(async cs => {
+          if (cs.commit) {
+            let { links } = await getInfo({
+              repo: options.repo,
+              commit: cs.commit
+            });
+            return links.commit;
+          }
+        })
+      )
+    )
+      .filter(_ => _)
+      .join(', ')}]:`;
+
+    const updatedDepenenciesList = dependenciesUpdated.map(
+      dependency => `  - ${dependency.name}@${dependency.newVersion}`
+    );
+
+    return [changesetLink, ...updatedDepenenciesList].join('\n');
+  },
+  getReleaseLine: async (changeset, type, options) => {
+    if (!options || !options.repo) {
+      throw new Error(
+        'Please provide a repo to this changelog generator like this:\n"changelog": ["@changesets/changelog-github", { "repo": "org/repo" }]'
+      );
+    }
+    const [firstLine, ...futureLines] = changeset.summary
+      .split('\n')
+      .map(l => l.trimRight());
+
+    if (changeset.commit) {
+      let { pull: pullNumber, links } = await getInfo({
+        repo: options.repo,
+        commit: changeset.commit
+      });
+
+      let fixedIssueLink = null;
+      // If the summary didn't mention any issue, we will look at the PR body to try to generate one automatically
+      if (!/issues\/[\d+]/i.test(changeset.summary) && pullNumber) {
+        fixedIssueLink = await getFixedIssueLink(pullNumber, options.repo);
+      }
+
+      return `\n\n- ${links.commit}${
+        links.pull === null ? '' : ` ${links.pull}`
+      }${
+        fixedIssueLink === null ? '' : ` ${fixedIssueLink}`
+      } - ${firstLine}\n${futureLines.map(l => `  ${l}`).join('\n')}`;
+    } else {
+      return `\n\n- ${firstLine}\n${futureLines.map(l => `  ${l}`).join('\n')}`;
+    }
+  }
+};
+
+const fixedIssueRegex = /(close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved) [^\s]*(#|issues\/)([\d]+)/i;
+async function getFixedIssueLink(
+  prNumber: number,
+  repo: string
+): Promise<string> {
+  const { body }: { body: string } = await fetch(
+    `https://api.github.com/repos/${repo}/pulls/${prNumber}?access_token=${process.env.GITHUB_TOKEN}`,
+    {
+      method: 'GET'
+    }
+  ).then(data => data.json());
+
+  const match = fixedIssueRegex.exec(body);
+  if (!match) {
+    return '';
+  }
+  const issueNumber = match[3];
+  return `Fixed [issue #${issueNumber}](https://github.com/firebase/firebase-js-sdk/issues/${issueNumber})`;
```

Comment:
Could we do something like `(fixes [#${issueNumber}])`? It's a bit shorter and conveys the same information.

Reply:
I did that initially, but thought people may confuse the PR number with the issue number and adding string `issue` makes it a bit clearer.

Otherwise the generated line would look like this ` #3224 fixes #3158`. Do you find it clear or confusing?

Reply:
I am very accustomed to this style, so it is not confusing to me. I do think the parenthesis might help.

We could also add it to the end:

```
9c409ea74 #3224 - [fix] Updated the TypeScript types for all APIs using Observers to allow callback omission (issue #3158).
```

I leave this up to your capable hands :)

Reply:
Ah, I didn't see the parenthesis. It certainly helps. Will adopt your suggestion. 
I don't like putting it in the end because it's less discoverable, especially when the description is long.